framework/src/actions/MaterialOutputAction.C

```diff
@@ -200,6 +204,11 @@ MaterialOutputAction::act()
     for (const auto & var_name : material_names)
     {
       oss << "\n  " << var_name;
+      // Warn if adding material properties auxiliary variables with the same name as a variable
+      if (_problem->hasVariable(var_name))
```

Comment:
I haven't really followed this closely, but I'm having trouble understanding why this is a warning rather than an error. It looks like if it already exists, we create it anyway (which I guess works), and then have two things potentially setting its value. Why not just make that an error?

Reply:
because the truss tests rely on it. When someone fixes them, I raised an issue, we can convert to an error. 
A warning is better than nothing, the current behavior.

Reply:
Why can we not fix those tests?

Reply:
I too would prefer to get the error in on the initial PR instead of having to do it in the future

Reply:
I've looked into changing and if we make it an error then we wont be able to recover simulations that have a material output. The first run will create the variables, the second one will error out on already existing variables. We could have people add a prefix in between, but it's not worth erroring out their simulation tbh

Reply:
looks like the PR is borked anyway. I ll get around to fixing it